docs/source/usage.rst

```diff
@@ -850,19 +852,22 @@ OmegaConf.select() allow you to select a config node or value using a dot-notati
 
 OmegaConf.update
 ^^^^^^^^^^^^^^^^
-OmegaConf.update() allow you to update values in your config using a dot-notation key.
+OmegaConf.update() allows you to update values in your config using either a dot-notation or brackets to denote sub-keys.
 
 The merge flag controls the behavior if the input is a dict or a list. If it's true, those are merged instead of
 being assigned.
 
 .. doctest::
 
     >>> cfg = OmegaConf.create({"foo" : {"bar": 10}})
-    >>> OmegaConf.update(cfg, "foo.bar", 20, merge=True) # merge has no effect because the value is a primitive
+    >>> # Merge flag has no effect because the value is a primitive
+    >>> OmegaConf.update(cfg, "foo.bar", 20, merge=True)
     >>> assert cfg.foo.bar == 20
-    >>> OmegaConf.update(cfg, "foo.bar", {"zonk" : 30}, merge=False) # set   
+    >>> # Set dictionary value (using dot notation)
+    >>> OmegaConf.update(cfg, "foo.bar", {"zonk" : 30}, merge=False)
     >>> assert cfg.foo.bar == {"zonk" : 30}
-    >>> OmegaConf.update(cfg, "foo.bar", {"oompa" : 40}, merge=True) # merge
+    >>> # Merge dictionary value (using bracket notation)
+    >>> OmegaConf.update(cfg, "foo[bar]", {"oompa" : 40}, merge=True)
```

Comment:
This is intertwining the merge and the key syntax.
Maybe something like:
```
    >>> # Merge flag has no effect because the value is a primitive
    >>> # Examples with different keys (no merge flag)
    >>> ...
    >>> # Examples with merge flag (one key style)
    >>> ...
```

Reply:
What about e54a860?
I moved the first example in last position as it seemed easier to understand after seeing what merge=True usually does.

Reply:
Looking at this, I prefer your previous version.
Let's go with that.

Reply:
Ok, so I'm reverting e54a86026fac57483dafa998e5c7a69fa3463909 then merging once all green